### PR TITLE
Reorganize GithubInfo componment

### PR DESCRIPTION
**Pull request by JiaHua-Zou**
## Issue This PR Addresses
#2380 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

The GitHub components are mixed with other files in the folder `.\src\web\src\components\Posts` and it's starting to look messy. Move all the GitHub components to the folder `.\src\web\src\components\Posts\GitHubInfo` to organize it.

## Steps to test the PR
1. `pnpm install`
2. `pnpm services:start`
3. `pnpm start`
4. Go to `localhost:8000`
5. Each Post should still have their GitHub components.

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Review by humphd — CHANGES_REQUESTED** [review]
Rename `src/web/src/components/Posts/GitHubInfo/GitHubInfo.tsx` to  `src/web/src/components/Posts/GitHubInfo/index.tsx`

File: src/web/src/components/Posts/GitHubInfo/Post.tsx [hunk 1]
```diff
@@ -16,16 +16,16 @@ import {
...
 import ErrorRoundedIcon from '@material-ui/icons/ErrorRounded';
 import LiteYouTubeEmbed from 'react-lite-youtube-embed';
-import GithubInfoProvider from '../GithubInfoProvider';
-import { Post } from '../../interfaces';
-import AdminButtons from '../AdminButtons';
-import Spinner from '../Spinner';
-import PostDesktopInfo from './PostInfo';
-import PostAvatar from './PostAvatar';
+import GithubInfoProvider from '../../GithubInfoProvider';
+import { Post } from '../../../interfaces';
```
Comment (DukeManh) [inline src/web/src/components/Posts/GitHubInfo/Post.tsx:20]: nit: Organize imports to components first and interfaces after?

Comment (JiaHua-Zou) [reply src/web/src/components/Posts/GitHubInfo/Post.tsx]: I reorganize the imports.

**Review by humphd — CHANGES_REQUESTED** [review]
`src/web/src/components/Posts/GitHubInfo/Index.tsx` is wrong, it should be `src/web/src/components/Posts/GitHubInfo/index.tsx`, lower case, and then you can import the folder name without any file within it.

**Review by DukeManh — DISMISSED** [review]

File: src/web/src/components/Posts/Timeline.tsx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { ReactElement } from 'react';
 import { Container, createStyles } from '@material-ui/core';
 import { makeStyles, Theme } from '@material-ui/core/styles';
-import PostComponent from './Post';
+import PostComponent from './GitHubInfo/Post';
```
Comment (DukeManh) [inline src/web/src/components/Posts/Timeline.tsx:4]: Why do you want  to move Post into GitHubInfo?

**Review by DukeManh — CHANGES_REQUESTED** [review]

**Comment by JiaHua-Zou** [comment]
For some reason git did not notice the uppercase change for the file `Index.tsx`. I had to change the filename to something. commit it, change it to the proper name (`index.tsx`) then I can finally commit and rebase it.

**Comment by DukeManh** [comment]
@JiaHua-Zou the index file is not a component so don't capitalize the name and use the `.tsx` extension.

File: src/web/src/components/Posts/Timeline.tsx (hunk 2 above)
Comment (JiaHua-Zou) [reply src/web/src/components/Posts/Timeline.tsx:4]: I was referring to different PR (#2379, #2367, #2376) and finding what files to move into the `GithubInfo` file. Should have not include that there.

**Comment by Andrewnt219** [comment]
@JiaHua-Zou it's case insensitive. Try removing the cache and add them all in again

```bash
git rm --cached path/to/Index.tsx
git add path/to/index.tsx
```

File: src/web/src/components/Posts/Timeline.tsx (hunk 2 above)
Comment (JiaHua-Zou) [reply src/web/src/components/Posts/Timeline.tsx]: ok. I move the file back

File: src/web/src/components/Posts/Post.tsx [hunk 3]
```diff
@@ -16,16 +16,16 @@ import {
...
 import ErrorRoundedIcon from '@material-ui/icons/ErrorRounded';
 import LiteYouTubeEmbed from 'react-lite-youtube-embed';
-import GithubInfoProvider from '../GithubInfoProvider';
-import { Post } from '../../interfaces';
-import AdminButtons from '../AdminButtons';
-import Spinner from '../Spinner';
-import PostDesktopInfo from './PostInfo';
-import PostAvatar from './PostAvatar';
-import GitHubInfo from './GitHubInfo';
+import GitHubInfo from './index';
```
Comment (DukeManh) [inline src/web/src/components/Posts/Post.tsx:19]:
```suggestion
import GitHubInfo from './GitHubInfo';
```

**Review by DukeManh — CHANGES_REQUESTED** [review]

File: src/web/src/components/Posts/Post.tsx (hunk 3 above)
Comment (JiaHua-Zou) [reply src/web/src/components/Posts/Post.tsx:19]: Is this to include all of the files in the `GitHubInfo` file?

Comment (DukeManh) [reply src/web/src/components/Posts/Post.tsx:19]: GitHubInfo folder has an index file so importing from `./GitHubInfo`  will resolve to the index file.

**Comment by JiaHua-Zou** [comment]
Had to push the branch again due to having the wrong import path.

File: src/web/src/components/Posts/Post.tsx [hunk 4]
```diff
@@ -16,16 +16,16 @@ import {
 } from '@material-ui/core';
 import ErrorRoundedIcon from '@material-ui/icons/ErrorRounded';
 import LiteYouTubeEmbed from 'react-lite-youtube-embed';
+import GitHubInfo from './GitHubInfo/index';
```
Comment (humphd) [inline src/web/src/components/Posts/Post.tsx:19]:
```ts
import GitHubInfo from './GitHubInfo';
```

**Review by humphd — CHANGES_REQUESTED** [review]

File: src/web/src/components/Posts/Post.tsx (hunk 3 above)
Comment (JiaHua-Zou) [reply src/web/src/components/Posts/Post.tsx]: ok got it. Thank.

**Review by humphd — APPROVED** [review]

**Review by HyperTHD — APPROVED** [review]
LGTM! Nice work

**Pull request merged**